### DISCUSSION TITLE
Force link re-draw

### DISCUSSION THREAD

**Discussion by dmb0058** [clientIO/joint#2563] [Q&A] [answered]
### Introduction

I'm sure this must be easier than I'm making it ...

I have a graph with elements and links. Each element has an output port that can be used to drag a link to another element.

When the link is connected I would like to have it removed from the output port and connected to the source body in the default position (i.e. where it would be if I had created the link and set its source programatically.

To do this I have used this code:
```
paper.on('link:connect', (linkView) => {
        let currentLink = linkView.model;
        
        // disconnect the link from the output port
        delete currentLink.attributes.source.port;
        delete currentLink.attributes.source.magnet;

       >>> somehow force the link to be redrawn here <<<
    });
```
I can't work out what I need to put to force the redraw - any ideas?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
1. you need to use one of the methods to set attributes on models: `set()`, `prop()`, `attrs()`.
In this case, you can also use `source()` and `target()` methods as explained here: https://resources.jointjs.com/tutorial/links
2. if you want the port and magnet to be used during connection, but not saved to the model, please check out [connectionStrategy](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#connectionStrategies.custom) instead.

**Reply by dmb0058** [reply]
Hmm, I'm doing something wrong then as I had tried `currentLink.attr('line/stroke', 'orange');` at the indicated position but it didn't do anything.

The connectionStrategy looks like exactly what I need, but as I read it, this function is called when the new link is connected into the model, and returns the end point to connect the link to ... how can I change the start point?

**Reply by kumilingus** [reply, marked as answer]
It was meant to be general. If you want to change any property of the model, you have to use the provided API (don't mutate the object)
```js
paper.on('link:connect', (linkView) => {
    let currentLink = linkView.model;
    // disconnect the link from the output port
    currentLink.source({ id: currentLink.source().id });
});
```

**Reply by dmb0058** [reply]
Thanks so much, I was completely over-thinking this. "... use the provided API (don't mutate the object) ..." is great advice, thanks, I'll remember that :)
